### PR TITLE
Calendar implementation beginning

### PR DESCRIPTION
**Pull request by brennanmcmicking**

### CODE REVIEW

File: src/middlewares.rs [hunk 1]
```diff
@@ -32,16 +28,16 @@ pub async fn parse_cookie(
...
-        let blank_selection: Vec<Course> = vec![Course {
-            name: "csc111".to_string(),
-            crns: Vec::new(),
-        }];
-
+        let blank_selection: Vec<ThinCourse> = Vec::new();
+        // blank_selection.push(Course {
+        //     name: "csc111".to_string(),
+        //     crns: Vec::new(),
+        // });
```
Comment (malcolmseyd) [inline src/middlewares.rs]:
```suggestion
```

File: src/routes/root.rs [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
-                .selection
-                .clone()
-                .iter()
-                .filter(|course| course.name == **x)
-                .count()
-                == 0
-        })
-        .map(|course| course.to_owned())
-        .collect();
+    // dbg!(&state);
```
Comment (malcolmseyd) [inline src/routes/root.rs]:
```suggestion
```

File: src/routes/term.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    //         user_state
+    //             .selection
+    //             .clone()
+    //             .iter()
+    //             .filter(|course| course.name == **x)
+    //             .count()
+    //             == 0
+    //     })
+    //     .map(|course| course.to_owned())
+    //     .collect();
```
Comment (malcolmseyd) [inline src/routes/term.rs]:
```suggestion
```

File: src/components/calendar.rs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+use maud::{html, Markup};
+
+pub fn c() -> Markup {
+    let timeslots: Vec<f64> = Vec::from_iter((17..46).map(|n| n as f64 * 0.5));
```
Comment (malcolmseyd) [inline src/components/calendar.rs:6]: I'm not really sure what that range is, could you elaborate or at least put it in a named variable?

File: src/components/calendar.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
+use maud::{html, Markup};
+
+pub fn c() -> Markup {
```
Comment (malcolmseyd) [inline src/components/calendar.rs]: Could we pick a better name for this? Not a huge fan of `c`.

**Review by malcolmseyd — DISMISSED** [review]
Looks good for the most part, just a few suggestions for clarity.

File: src/components/calendar.rs (hunk 5 above)
Comment (brennanmcmicking) [reply src/components/calendar.rs]: yes, how about `render`?

File: src/components/calendar.rs (hunk 4 above)
Comment (brennanmcmicking) [reply src/components/calendar.rs:6]: Yes, it's supposed to build a vec from 8.5 -> 23.0 at increments of 0.5 to represent the grid on the calendar. I will add a comment to make this clearer (though I'm not sure I'm going to implement the calendar this way; I'm taking a design-as-I-code approach).

File: src/components/calendar.rs (hunk 5 above)
Comment (malcolmseyd) [reply src/components/calendar.rs]: That would work, you could also do `calendar` or `render_calendar`. Something less vague, that's all.

Comment (brennanmcmicking) [reply src/components/calendar.rs]: When looking at [the way that this function will be called](https://github.com/brennanmcmicking/scheduler/blob/4a6752e0d60e347f5917b60d9fdcd2fa481d9441/src/components/container.rs#L23), I think it makes more sense to name it `render` so the line in the example I linked would look like
```
(components::search_result::render(...))
```

Comment (malcolmseyd) [reply src/components/calendar.rs]: Okay, that makes sense. In that case I guess the name doesn't really matter.

**Review by malcolmseyd — APPROVED** [review]
LGTM!!!!!

**Pull request merged**